### PR TITLE
Allow creating IPAddress object from string and provide = overload

### PR DESCRIPTION
**Pull request by pennam**

### CODE REVIEW

**Review by aentinger — APPROVED** [review]
LGTM :+1:

**Comment by pennam** [comment]
Fixed typo in commit message

File: api/IPAddress.h [hunk 1]
```diff
@@ -67,6 +67,8 @@ class IPAddress : public Printable {
      // Default IPv4
     IPAddress(const uint8_t *address);
     IPAddress(IPType ip_type, const uint8_t *address);
+    // If IPv4 fails tries IPv6 see fromSting function
```
Comment (per1234) [inline api/IPAddress.h:70]:
```suggestion
    // If IPv4 fails tries IPv6 see fromString function
```

File: api/IPAddress.h [hunk 2]
```diff
@@ -90,6 +92,8 @@ class IPAddress : public Printable {
     IPAddress& operator=(const uint8_t *address);
     // NOTE: IPv4 only; see implementation note
     IPAddress& operator=(uint32_t address);
+    // If IPv4 fails tries IPv6 see fromSting function
```
Comment (per1234) [inline api/IPAddress.h:95]:
```suggestion
    // If IPv4 fails tries IPv6 see fromString function
```

**Comment by pennam** [comment]
1 commit 3 typos :sob: thanks @per1234

**Review by per1234 — APPROVED** [review]

**Pull request merged**